polkadot/runtime/parachains/src/hrmp.rs

```diff
@@ -640,6 +659,131 @@ pub mod pallet {
+			let max_capacity = config.hrmp_channel_max_capacity;
+
+			Self::init_open_channel(sender, recipient, max_capacity, max_message_size)?;
+			Self::accept_open_channel(recipient, sender)?;
+
+			Self::deposit_event(Event::HrmpSystemChannelOpened(
+				sender,
+				recipient,
+				max_capacity,
+				max_message_size,
+			));
+
+			Ok(Pays::No.into())
+		}
+
+		/// Update the deposits held for an HRMP channel. If at least one of the chains is a system
+		/// chain, any deposits held will be unreserved.
+		///
+		/// Arguments:
+		///
+		/// - `sender`: A chain, `ParaId`.
+		/// - `recipient`: A chain, `ParaId`.
+		///
+		/// Any signed origin can call this function.
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::poke_channel_deposits())]
+		pub fn poke_channel_deposits(
+			origin: OriginFor<T>,
+			sender: ParaId,
+			recipient: ParaId,
+		) -> DispatchResult {
+			let _caller = ensure_signed(origin)?;
+			let channel_id = HrmpChannelId { sender, recipient };
+			let is_system = sender.is_system() || recipient.is_system();
+
+			let config = <configuration::Pallet<T>>::config();
+
+			// Channels with and amongst the system do not require a deposit.
+			let (new_sender_deposit, new_recipient_deposit) = if is_system {
+				(0, 0)
+			} else {
+				(config.hrmp_sender_deposit, config.hrmp_recipient_deposit)
+			};
+
+			let _ = HrmpChannels::<T>::mutate(&channel_id, |channel| -> DispatchResult {
+				if let Some(ref mut channel) = channel {
+					let current_sender_deposit = channel.sender_deposit;
+					let current_recipient_deposit = channel.recipient_deposit;
+
+					// nothing to update
+					if current_sender_deposit == new_sender_deposit &&
+						current_recipient_deposit == new_recipient_deposit
+					{
+						return Ok(())
+					}
+
+					// sender
+					if current_sender_deposit > new_sender_deposit {
+						T::Currency::unreserve(
+							&channel_id.sender.into_account_truncating(),
+							(current_sender_deposit - new_sender_deposit).unique_saturated_into(),
+						);
+					} else if current_sender_deposit < new_sender_deposit {
+						T::Currency::reserve(
+							&channel_id.sender.into_account_truncating(),
+							(new_sender_deposit - current_sender_deposit).unique_saturated_into(),
+						)?;
+					}
+
+					// recipient
+					if current_recipient_deposit > new_recipient_deposit {
+						T::Currency::unreserve(
+							&channel_id.recipient.into_account_truncating(),
+							(current_recipient_deposit - new_recipient_deposit)
+								.unique_saturated_into(),
+						);
+					} else if current_recipient_deposit < new_recipient_deposit {
+						T::Currency::reserve(
+							&channel_id.recipient.into_account_truncating(),
+							(new_recipient_deposit - current_recipient_deposit)
+								.unique_saturated_into(),
+						)?;
+					}
+
+					// update storage
+					channel.sender_deposit = new_sender_deposit;
+					channel.recipient_deposit = new_recipient_deposit;
+				}
+				Ok(())
+			});
```

Comment:
```suggestion
			})?;
```
otherwise if currency reserve, or anything else fail with error, it wont bubble the error up. it will be successful.